# Task no 1

In [ ]:
import pandas as pd
data=pd.read_csv('pubg - Dr. Darshan Ingle.csv')

# Task no 2

In [4]:
data.dtypes

Id                  object
groupId             object
matchId             object
assists              int64
boosts               int64
damageDealt        float64
DBNOs                int64
headshotKills        int64
heals                int64
killPlace            int64
killPoints           int64
kills                int64
killStreaks          int64
longestKill        float64
matchDuration        int64
matchType           object
maxPlace             int64
numGroups            int64
rankPoints           int64
revives              int64
rideDistance       float64
roadKills            int64
swimDistance       float64
teamKills            int64
vehicleDestroys      int64
walkDistance       float64
weaponsAcquired      int64
winPoints            int64
winPlacePerc       float64
dtype: object

# Task no 3

In [6]:
data.describe()

assists        boosts   damageDealt        DBNOs  headshotKills  \
count  10000.000000  10000.000000  10000.000000  10000.00000   10000.000000   
mean       0.234600      1.088500    129.211264      0.64400       0.221700   
std        0.575149      1.703279    167.193945      1.09562       0.577046   
min        0.000000      0.000000      0.000000      0.00000       0.000000   
25%        0.000000      0.000000      0.000000      0.00000       0.000000   
50%        0.000000      0.000000     83.805000      0.00000       0.000000   
75%        0.000000      2.000000    185.325000      1.00000       0.000000   
max        7.000000     18.000000   3469.000000     11.00000      14.000000   

              heals     killPlace    killPoints         kills   killStreaks  \
count  10000.000000  10000.000000  10000.000000  10000.000000  10000.000000   
mean       1.354000     47.663100    506.970200      0.913400      0.543800   
std        2.629102     27.424146    627.297959      1.524117      0.701948   
min        0.000000      1.000000      0.000000      0.000000      0.000000   
25%        0.000000     24.000000      0.000000      0.000000      0.000000   
50%        0.000000     48.000000      0.000000      0.000000      0.000000   
75%        2.000000     71.000000   1169.000000      1.000000      1.000000   
max       31.000000    100.000000   1926.000000     35.000000      4.000000   

       ...       revives  rideDistance     roadKills  swimDistance  \
count  ...  10000.000000  10000.000000  10000.000000  10000.000000   
mean   ...      0.160200    600.693584      0.004200      4.385917   
std    ...      0.454045   1524.915601      0.074719     30.889620   
min    ...      0.000000      0.000000      0.000000      0.000000   
25%    ...      0.000000      0.000000      0.000000      0.000000   
50%    ...      0.000000      0.000000      0.000000      0.000000   
75%    ...      0.000000      0.000575      0.000000      0.000000   
max    ...      5.000000  28780.000000      3.000000    971.200000   

          teamKills  vehicleDestroys  walkDistance  weaponsAcquired  \
count  10000.000000     10000.000000  10000.000000      10000.00000   
mean       0.024400         0.007700   1130.008410          3.63590   
std        0.171486         0.089674   1168.597983          2.42209   
min        0.000000         0.000000      0.000000          0.00000   
25%        0.000000         0.000000    151.575000          2.00000   
50%        0.000000         0.000000    650.350000          3.00000   
75%        0.000000         0.000000   1923.250000          5.00000   
max        3.000000         2.000000  10490.000000         41.00000   

        winPoints  winPlacePerc  
count  10000.0000  10000.000000  
mean     609.3440      0.469926  
std      739.7924      0.304508  
min        0.0000      0.000000  
25%        0.0000      0.200000  
50%        0.0000      0.458300  
75%     1495.0000      0.735100  
max     1863.0000      1.000000  

[8 rows x 25 columns]

# Task no 4

In [7]:
data.kills.mean()

0.9134

# Task no 5

In [9]:
import numpy as np
np.percentile(data.kills,99)

7.0

# Task no 6

In [11]:
data.kills.max()

35

# Task no 7

In [12]:
data.columns

Index(['Id', 'groupId', 'matchId', 'assists', 'boosts', 'damageDealt', 'DBNOs',
       'headshotKills', 'heals', 'killPlace', 'killPoints', 'kills',
       'killStreaks', 'longestKill', 'matchDuration', 'matchType', 'maxPlace',
       'numGroups', 'rankPoints', 'revives', 'rideDistance', 'roadKills',
       'swimDistance', 'teamKills', 'vehicleDestroys', 'walkDistance',
       'weaponsAcquired', 'winPoints', 'winPlacePerc'],
      dtype='object')

# Task no 8

#### whenever i try to plot the graph my kernel dies. So, in this project i didn't shown the plotting of graph instead have only written the code for it.

In [ ]:
import pandas as pd
data=pd.read_csv('pubg - Dr. Darshan Ingle.csv')
import seaborn as sns
sns.displot(data.matchDuration);

1. min value is 464 2. max values is 2202 3. std is 258.96 4. mean is 1575.39

# Task no 9

In [ ]:
import pandas as pd
data=pd.read_csv('pubg - Dr. Darshan Ingle.csv')
import seaborn as sns
sns.distplot(data['walkDistance']);

1. min value is 0 2. max values is 10490 3. std is 1168.59 4. mean is 1130.00

# Task no 10

In [ ]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
data=pd.read_csv('pubg - Dr. Darshan Ingle.csv')
fig,axs=plt.subplots(2,1)
sns.histplot(data.matchDuration,ax=axs[0]);
sns.histplot(data.walkDistance,ax=axs[1]);


# Task no 11

In [ ]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
data=pd.read_csv('pubg - Dr. Darshan Ingle.csv')
fig,axs=plt.subplots(1,2)
sns.histplot(data.matchDuration,ax=axs[0]);
sns.histplot(data.walkDistance,ax=axs[1]);


# Task no 12

In [ ]:
import pandas as pd
import seaborn as sns
data=pd.read_csv('pubg - Dr. Darshan Ingle.csv')
sns.pairplot(data);

# Task no 13

In [1]:
import pandas as pd
data=pd.read_csv('pubg - Dr. Darshan Ingle.csv')
data.matchType.value_counts()

squad-fpp           3969
duo-fpp             2282
squad               1359
solo-fpp            1234
duo                  702
solo                 386
normal-squad-fpp      24
normal-duo-fpp        13
crashfpp              13
normal-solo-fpp        8
normal-squad           4
flaretpp               3
crashtpp               2
flarefpp               1
Name: matchType, dtype: int64

# Task no 14

In [ ]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
data=pd.read_csv('pubg - Dr. Darshan Ingle.csv')
sns.barplot(data.matchType,data.killPoints);
plt.xticks(rotation=70);

C:\Users\rohan\anaconda3\lib\site-packages\seaborn\_decorators.py:36: FutureWarning: Pass the following variables as keyword args: x, y. From version 0.12, the only valid positional argument will be `data`, and passing other arguments without an explicit keyword will result in an error or misinterpretation.
  warnings.warn(


normal-squad-fpp and normal-squad-fpp match types has the most kill points whereas solo-fpp and solo match types has the less kill points.

# Task no 15

In [ ]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
data=pd.read_csv('pubg - Dr. Darshan Ingle.csv')
sns.barplot(data.matchType,data.weaponsAcquired);
plt.xticks(rotation=70);

C:\Users\rohan\anaconda3\lib\site-packages\seaborn\_decorators.py:36: FutureWarning: Pass the following variables as keyword args: x, y. From version 0.12, the only valid positional argument will be `data`, and passing other arguments without an explicit keyword will result in an error or misinterpretation.
  warnings.warn(


In crashtpp and crashfpp match types, the weapons acquired by players are very less whereas in normal-solo-fpp and normal-squad-fpp match types,the weapons acquired by players are more.

# Task no 16

In [2]:
import pandas as pd
data=pd.read_csv('pubg - Dr. Darshan Ingle.csv')
data.select_dtypes(exclude=['number'])

Id         groupId         matchId  matchType
0     2f262dd9795e60  78437bcd91d40e  d5db3a49eb2955  squad-fpp
1     a32847cf5bf34b  85b7ce5a12e10b  65223f05c7fdb4  squad-fpp
2     1b1900a9990396  edf80d6523380a  1cadec4534f30a  squad-fpp
3     f589dd03b60bf2  804ab5e5585558  c4a5676dc91604      squad
4     c23c4cc5b78b35  b3e2cd169ed920  cd595700a01bfa  squad-fpp
...              ...             ...             ...        ...
9995  ef4f474acd8e85  2eca2a8391f75d  492ecdfae90b46  squad-fpp
9996  cf0bf82fb4d80e  2eaf2765f93adb  14bffd71e96320    duo-fpp
9997  a0a31a0b1dcbe1  8d50c64ccc5071  147e4bbb62e3bb    duo-fpp
9998  f6874657399d69  d31843d7e62ccb  662567dcf280f5    duo-fpp
9999  90359b0b8f8b0d  61d5b1bb8da43f  258bfa48d88014       solo

[10000 rows x 4 columns]

# Task no 17

In [ ]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
data=pd.read_csv('pubg - Dr. Darshan Ingle.csv')
sns.boxplot(data.matchType,data.winPlacePerc);
plt.xticks(rotation=70);

The flarefpp and flaretpp has lowest winPlacePerc whereas the normal-duo-fpp and normal-squad-tpp has highest winPlacePerc.

# Task no 18

In [ ]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
data=pd.read_csv('pubg - Dr. Darshan Ingle.csv')
sns.boxplot(data.matchType,data.matchDuration);
plt.xticks(rotation=70);

The flarefpp,flaretpp,crashfpp and crashtpp has less matchDuration whereas normal-squad and normal-squad-fpp has more matchDuration.

# Task no 19

In [ ]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
data=pd.read_csv('pubg - Dr. Darshan Ingle.csv')
sns.boxplot(data.matchDuration,data.matchType);
plt.xticks(rotation=70);

C:\Users\rohan\anaconda3\lib\site-packages\seaborn\_decorators.py:36: FutureWarning: Pass the following variables as keyword args: x, y. From version 0.12, the only valid positional argument will be `data`, and passing other arguments without an explicit keyword will result in an error or misinterpretation.
  warnings.warn(


# Task no 20

In [3]:
import pandas as pd
data=pd.read_csv('pubg - Dr. Darshan Ingle.csv')
data['kill']=data['headshotKills']+data['roadKills']+data['teamKills']
data.head()

Id         groupId         matchId  assists  boosts  \
0  2f262dd9795e60  78437bcd91d40e  d5db3a49eb2955        0       0   
1  a32847cf5bf34b  85b7ce5a12e10b  65223f05c7fdb4        0       0   
2  1b1900a9990396  edf80d6523380a  1cadec4534f30a        0       3   
3  f589dd03b60bf2  804ab5e5585558  c4a5676dc91604        0       0   
4  c23c4cc5b78b35  b3e2cd169ed920  cd595700a01bfa        0       0   

   damageDealt  DBNOs  headshotKills  heals  killPlace  ...  rideDistance  \
0          0.0      0              0      0         92  ...           0.0   
1        163.2      1              1      0         42  ...           0.0   
2        278.7      2              1      8         16  ...           0.0   
3        191.9      1              0      0         31  ...           0.0   
4        100.0      1              0      0         87  ...           0.0   

   roadKills  swimDistance  teamKills  vehicleDestroys walkDistance  \
0          0           0.0          0                0          0.0   
1          0           0.0          0                0        132.7   
2          0           0.0          0                0       3591.0   
3          0           0.0          0                0        332.7   
4          0           0.0          0                0        252.7   

   weaponsAcquired  winPoints  winPlacePerc  kill  
0                0       1470        0.0000     0  
1                2       1531        0.2222     1  
2               10          0        0.8571     1  
3                3          0        0.3462     0  
4                3       1557        0.0690     0  

[5 rows x 30 columns]

# Task no 21

In [4]:
import pandas as pd
data=pd.read_csv('pubg - Dr. Darshan Ingle.csv')
data['winPlacePerc']=round(data['winPlacePerc'],2)
data['winPlacePerc']

0       0.00
1       0.22
2       0.86
3       0.35
4       0.07
        ... 
9995    0.83
9996    0.72
9997    0.21
9998    0.24
9999    0.19
Name: winPlacePerc, Length: 10000, dtype: float64

# Task no 22

In [ ]:
import numpy as np
import pandas as pd
import seaborn as sns
data=pd.read_csv('pubg - Dr. Darshan Ingle.csv')
x=[]
for i in range(100):
    x.append(data['damageDealt'].sample(50).mean())
means = np.array(x)
sns.histplot(means);

Most of the damagedealt is in the range of 150 while the least damagedealt is in 200